network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/waitlist/NetPWaitlistManager.kt

```diff
@@ -55,17 +57,29 @@ class RealNetPWaitlistManager @Inject constructor(
     private val networkProtectionWaitlist: NetworkProtectionWaitlist,
     private val netPWaitlistService: NetPWaitlistService,
     private val dispatcherProvider: DispatcherProvider,
+    @AppCoroutineScope private val coroutineScope: CoroutineScope,
 ) : NetPWaitlistManager {
 
-    private val state: MutableStateFlow<NetPWaitlistState> =
-        MutableStateFlow(networkProtectionWaitlist.getState())
+    // A state flow behaves identically to a shared flow when it is created with the following parameters
+    // See https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines.flow/-state-flow/
+    // See also https://github.com/Kotlin/kotlinx.coroutines/issues/2515
+    private val state: MutableSharedFlow<NetPWaitlistState> = MutableSharedFlow(
```

Comment:
@aitorvs assuming that it takes awhile before `getState()` emits the value, what would happen to the consumers of this flow?

Reply:
Nothing, they'd wait for that value to be emitted.